code/datums/stars.dm

```diff
@@ -290,3 +286,88 @@
 		P.disp_dist = 0.25 * (4 ** (P.disp_level/planets.len))*1.3333 - 0.3333
 		P.disp_x = cos(P.disp_angle) * P.disp_dist
 		P.disp_y = sin(P.disp_angle) * P.disp_dist
+
+/datum/station_module
+	var/list/buy_keywords = list()		// Associated list of keywords and price modifiers. Price modifiers are multiplicative.
+	var/list/sell_keywords = list()		// As above, but for players selling to the station.
+	var/rarity = 50		// weight of a module to get picked.
+	var/datum/space_station/station
+	var/module_suffix = "Meta-Physical Object That Should Not Exist"		//suffix of a station. Flavortext!
+	var/name		//actual constructed name
+/datum/station_module/New(var/datum/space_station/S)
+	if(S)
+		station = S
+	if(station)
+		buy_keywords[station.planet.parent_system.alignment] = 1 // faction is gotten for chance modifiers; actual faction price adjustment is done in recalculate_prices
+	if(!name)
+		name = "[new_station_name()] ([module_suffix])"
+
+/datum/station_module/toys
+	buy_keywords = list("Ammo" = 1.25, "Toys" = 0.75, "Security" = 1.5)
+	sell_keywords = list()
+	module_suffix = "Toy Factory"
+
+/datum/station_module/weaponry
+	buy_keywords = list("Ammo"= 0.75, "Security" = 0.75, "Clothing"=  1, "Melee" = 1)
+	sell_keywords = list("Security" = 0.5, "Material" = 1.25, "Salvage" = 1.25, "Robotics" = 1.5)
+	module_suffix = "Arms Dealer"
+	rarity = 30
+
+/datum/station_module/emergency
+	buy_keywords = list("Emergency" = 0.5, "Engineering" = 1, "Atmos" = 1)
+	sell_keywords = list("Emergency" = 0.5)
+	module_suffix = "First Responder Outpost"
+
+/datum/station_module/medical
+	buy_keywords = list("Medical" = 0.75, "Food" = 1, "Security" = 1.5)
+	sell_keywords = list("Medical" = 0.75)
+	module_suffix = "Space Hospital"
+	rarity = 30
+
+/datum/station_module/pizza_hut
+	buy_keywords = list("Pizza"=1, "Food"=0.9)
+	rarity = 20
+
+/datum/station_module/pizza_hut/New()
+	name = pick("Pizza Mutt", "Father Johnson's", "Pete's Pizza Parlor")
+	.=..()
```

Comment:
`. = ..()` comes first

Reply:
Doing this before the super avoids calling new_station_name() and then overriding it

Reply:
a proper way of overriding names in station_modules would be better then